esi_leap/objects/offer.py

```diff
@@ -88,3 +111,24 @@ def expire(self, context=None):
         # expire offer
         self.status = statuses.EXPIRED
         self.save(context)
+
+    def to_marketplace_dict(self):
+        # change fields name
+        offer_dict = self.to_dict()
+        offer_dict['start_time'] = offer_dict.pop('start_date').isoformat()
+        offer_dict['end_time'] = offer_dict.pop('end_date').isoformat()
+        offer_dict['cost'] = offer_dict['properties'].get('floor_price', 0)
+        offer_dict['server_config'] = offer_dict.pop('properties')
+        offer_dict['server_id'] = offer_dict.pop('resource_uuid')
+        offer_dict['provider_id'] = offer_dict.pop('uuid')
+        offer_dict['creator_id'] = offer_dict.pop('project_id')
+
+        # remove unnecessary feilds
+        offer_dict.pop('created_at')
+        offer_dict.pop('updated_at')
+        offer_dict.pop('id')
+        offer_dict.pop('resource_type')
+        # fake fields
+        offer_dict['marketplace_date_created'] = '2016-07-16T19:20:30'
```

Comment:
Can you set this to something like `datetime.datetime.utcnow()`?

Reply:
fixed